x-pack/legacy/plugins/ml/common/types/kibana.ts

```diff
@@ -17,8 +18,8 @@ export interface Route {
   k7Breadcrumbs: () => any;
 }
 
-export type IndexPatternSavedObject = SimpleSavedObject<SavedObjectAttributes>;
-export type SavedSearchSavedObject = SimpleSavedObject<SavedObjectAttributes>;
+export type IndexPatternSavedObject = SimpleSavedObject<IndexPatternAttributes>;
+export type SavedSearchSavedObject = SimpleSavedObject<any>;
```

Comment:
I think defining the shape of all these types are out of the scope of this PR, but it would be great if teams implemented these types to improve the type safety of their code. Maybe we can add a `// TODO: Define saved object type` everywhere where we have an `any`?

Reply:
done